### PR TITLE
continue reading section

### PR DESCRIPTION
**Pull request by coreyfloyd**
Pushing last read article if less than 24 hours and the app was quit.

And adding new section, which means:
- New section controller
- New cell
- New Schema item
- Updating the Schema to vend the section based on the rules (show after 24 hours)

Additionally added the support for icons on the sections

### CODE REVIEW

**Comment by coreyfloyd** [comment]
@montehurd @bgerstle ready to go - will do the close button separately so Monte can get my header work in and finish the styling.

**Comment by coreyfloyd** [comment]
@montehurd @bgerstle stalked with Josh on a few things:

- Punting on the close button for now. That will be resolved in a ticket for collapsing all sections
- Following this patch is another where I have enabled Facebook Tweaks for testing this feature (by allowing you to always show the continue reading section)

**Comment by bgerstle** [comment]
@coreyfloyd no need to stalk him... 😁

**Comment by bgerstle** [comment]
> Following this patch is another where I have enabled Facebook Tweaks for testing this feature (by allowing you to always show the continue reading section)

Did you ask him about making it an app setting? Tweaks is good for internal/debug stuff, but this could actually be a user-facing preference.

**Comment by coreyfloyd** [comment]
> Did you ask him about making it an app setting? Tweaks is good for internal/debug stuff, but this could actually be a user-facing preference.

No, I didn't can be brought up later in Product discussions if you like.

**Comment by coreyfloyd** [comment]
> @coreyfloyd no need to stalk him... 😁

I'm great at typos

**Comment by coreyfloyd** [comment]
screenshot 2015-10-08 15 40 27

**Comment by coreyfloyd** [comment]
Forgot the screenshot

**Comment by bgerstle** [comment]
@coreyfloyd are we showing the title in "Continue Reading"?

**Comment by coreyfloyd** [comment]
> @coreyfloyd are we showing the title in "Continue Reading"?

@bgerstle can you be more specific?

**Comment by bgerstle** [comment]
@coreyfloyd like "Continue Reading **Wolfdog**"

**Comment by bgerstle** [comment]
oh i see, it's **Dog** i'm going to continue reading. i think that separator needs to be thinner according to the mock

**Comment by coreyfloyd** [comment]
> oh i see, it's Dog i'm going to continue reading. i think that separator needs to be thinner according to the mock

@bgerstle right, had a note that Monte is finishing the styling there… 

>@montehurd @bgerstle ready to go - will do the close button separately so Monte can get my header work in and finish the styling.

Thats not really this ticket… I just did some to help him out and so I could put the icons in there. Sorry that wasn't clear.

**Comment by bgerstle** [comment]
i see, i didn't realize that's what the header styling was referring to

**Comment by bgerstle** [comment]
for the record: :-1: pending implementation of:

#### Preserve in-memory state when resumed w/in 24 hours
```
Given the app has been in the background for less than 24 hours
When I launch the app
Then I should see the previous state (i.e. no action taken)
```

```
Given was viewing article X
And the app was terminated less than 24 hours ago
When I launch the app
Then I should see article X (pushed on the Home tab for now)
```

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd updated the pull request to now only push the last "open" article - not the last read article.

FYI: 2 follow ons coming after this is merged:

One for adding the Facebook tweak to show the continue reading section for testing
Another that will enable the restore of the scroll position for an article after it is loaded

File: Wikipedia/UI-V5/WMFContinueReadingCell.m [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+- (UICollectionViewLayoutAttributes*)preferredLayoutAttributesFittingAttributes:(UICollectionViewLayoutAttributes*)layoutAttributes {
+    
+    CGFloat const preferredMaxLayoutWidth = layoutAttributes.size.width - self.leadingConstraint.constant - self.trailingConstraint.constant;
+
+    self.title.preferredMaxLayoutWidth = preferredMaxLayoutWidth;
+    self.summary.preferredMaxLayoutWidth = preferredMaxLayoutWidth;
+
+    //HACK: can't call [self.title systemLayoutSizeFittingSize:CGSizeMake(preferredMaxLayoutWidth, CGFLOAT_MAX)];
+    // This causes an arithmetic crash from some sort of semi-infinite loop through this method
```
Comment (montehurd) [inline Wikipedia/UI-V5/WMFContinueReadingCell.m]: Did you try:
[self.title systemLayoutSizeFittingSize:*UILayoutFittingCompressedSize*]

Comment (bgerstle) [inline Wikipedia/UI-V5/WMFContinueReadingCell.m]: also, if I'm understanding your intent correctly, you should try the following (as implemented in `UIView(VisualTestSizingUtils)`:

``` objc
    CGSize sizeThatFitsWidth = [self systemLayoutSizeFittingSize:preHeightAdjustmentSize
                                   withHorizontalFittingPriority:UILayoutPriorityRequired
                                         verticalFittingPriority:UILayoutPriorityFittingSizeLevel];
```

Comment (bgerstle) [inline Wikipedia/UI-V5/WMFContinueReadingCell.m]: if the above works, we could refactor the "expand height to fit `width`" logic into a `UIView` category in the app, which the `VisualTestSizingUtils` class reuses

**Comment by coreyfloyd** [comment]
@bgerstle addressed your comments… can't remove all the equality stuff from MWKTitle because we exclude the fragment there.

**Comment by bgerstle** [comment]
@coreyfloyd thanks, looks good :+1:! check out my comment on the `entryForTitle:` wrapper though

**Pull request merged**